std/math.d

```diff
@@ -1605,14 +1605,24 @@ real exp(real x) @trusted pure nothrow @nogc
         else
             static assert(0, "No over-/underflow limits for real type!");
 
-        // Special cases.
-        // FIXME: set IEEE flags accordingly
+        // Special cases. Raises an overflow or underflow flag accordingly,
+        // except in the case for CTFE, where there are no hardware controls.
         if (isNaN(x))
             return x;
         if (x > OF)
-            return real.infinity;
+        {
+            if (__ctfe)
+                return real.infinity;
+            else
+                return real.max * copysign(real.max, x);
```

Comment:
Slow how?  This operation is deliberate because I didn't 't want the optimizer to fold the operation so that nohardware flag is raised.